_posts/2020-08-07-g2p.md

```diff
@@ -392,40 +406,215 @@ You can also use abbreviations like this in the G2P studio by writing them in th
+
+#### Visualizing the process
+
+This is actually where the introductory picture to this blog post comes from! In the graphic below, you can see the word 'bonjour' in French converted to its IPA transcription. Then, its IPA transcription gets converted into its corresponding *English* IPA transcription based on the generated mapping (notice how uvular /ʁ/ gets transformed to alveolo-palatal /ʒ/). Then finally, the English IPA form is converted to Arpabet. So in all, we have an input of 'bonjour' that gets output as 'B AO N ZH UW ZH' and the aligner runs on that form.
+
+{% picture bonjour-g2p.png %}
+
+You can recreate this type of animation using the G2P studio by selecting the French (`fra`) to English Arpabet (`eng-arpabet`) mapping, choosing 'animate' and then typing your text in the input area as seen below:
+
+{% picture g2p-animate.png %}
+
+#### What can you do with your aligned audio/text?
+
+ReadAlongs exports to a variety of formats, including epub (for e-readers), [Praat](https://www.fon.hum.uva.nl/praat/) TextGrids, [ELAN](https://archive.mpi.nl/tla/elan) files, various subtitle formats, and an embeddable web component for your website. This project is still *very* unstable though, and it is **not** yet recommended to use it in production.
+
+Below is an example of the embeddable web component in Danish.
+
+{% include readalong.html text="/assets/ras/output.xml" alignment="/assets/ras/output.smil" audio="/assets/ras/output.wav" %}
 
 ### Convertextract
-[Convertextract]({{ "convertextract-app" | absolute_url }})
+
+Convertextract is a tool that lets you use `g2p` mappings to convert text inside a variety of different documents. Have a look at the [convertextract repo readme](https://github.com/roedoejet/convertextract) or at Fineen Davis' [blog post]({{ "convertextract-app" | absolute_url }}) about the Convertextract GUI.
 
 ### Advanced: use in Python
 
-...more to come...
+#### Installation
+
+We recommend installing `g2p` using [pip](https://en.wikipedia.org/wiki/Pip_(package_manager)):
+
+```python
+pip install g2p
+```
+
+If you are going to be creating your own mapping or editing `g2p` in any way, you must first fork and clone `g2p`:
+
+```bash
+git clone https://github.com/YourGitHubUsername/g2p.git
+```
+
+and then do an *editable* pip installation:
+
+```bash
+cd g2p && pip install -e .
+```
+
+#### Usage
+
+Using `g2p` within Python can be done, either programatically using the `make_g2p` function:
+
+```python
+>>> from g2p import make_g2p
+>>> transducer = make_g2p('dan', 'eng-arpabet')
+>>> transducer('hej').output_string
+'HH EH Y'
+```
+
+#### Command Line Interface
+
+You can also use `g2p` from the command line. The basic command for conversions is:
+
+```bash
+g2p convert <input_text> <in_lang> <out_lang>
+```
+
+So in practice:
+
+```bash
+$ g2p convert hej dan eng-arpabet
+HH EH YY
+```
 
 ## Advanced: contributing your rules to the main `g2p` library
```

Comment:
I think this should be its own, standalone post: "Tutorial: How to contribute new rules to `g2p`". I know I would have benefited from it a few weeks ago! 😂 

If you'd like, you can assign me to flesh-out this post. If you like this idea, I would cut this section out of the g2p post, and place it in `_drafts/tutorial-contributing-g2p-rules`.  If we publish this before October, it could be a nice tutorial for Hacktoberfest!

Reply:
ooo, I like the idea about Hactoberfest! and thanks, it would be great to have some help on that post, especially since it's fresh for you 😄

Reply:
I'd be glad to! Again, just dump what you have here in `_drafts/` and I can pick it up!